mountpoint-s3-crt/src/s3/client.rs

```diff
@@ -442,6 +461,26 @@ unsafe extern "C" fn meta_request_shutdown_callback(user_data: *mut libc::c_void
     std::mem::drop(user_data);
 }
 
+/// Safety: Don't call this function directly, only called by the CRT as a callback.
+unsafe extern "C" fn meta_request_upload_review_callback(
+    _request: *mut aws_s3_meta_request,
+    upload_review: *const aws_s3_upload_review,
+    user_data: *mut libc::c_void,
+) -> i32 {
+    // SAFETY: user_data always will be a MetaRequestOptionsInner since that's what we set it to
+    // in MetaRequestOptions::new.
+    let user_data = MetaRequestOptionsInner::from_user_data_ptr(user_data);
+
+    if let Some(callback) = user_data.on_upload_review.take() {
```

Comment:
The `return match` below confused me (I didn't notice we were returning the value of `aws_raise_error(err)`). Maybe write this as
```rust
let Some(callback) = user_data.on_upload_review.take() else {
    return AWS_OP_SUCCESS;
};
```
so the later control flow is clearer